src/Atmos/Model/prog_prim_conversion.jl

```diff
@@ -73,33 +56,25 @@ function prognostic_to_primitive!(
     moist::DryModel,
     prim::Vars,
     prog::Vars,
-    e_int::AbstractFloat,
+    aux::Vars,
 )
-    ts = PhaseDry(parameter_set(atmos), e_int, prog.ρ)
-    prim.ρ = prog.ρ
-    prim.u = prog.ρu ./ prog.ρ
-    prim.p = air_pressure(ts)
+    ts = new_thermo_state(atmos, prog, aux)
+    prim.ρ = air_density(ts) # Needed for recovery of energy, not prog.ρ in anelastic1d
```

Comment:
In the anelastic case (where we initiate state.ρ with aux_ref_state.ρ and state.ρ has no tendencies) do you expect to have any difference between state.ρ and aux_ref_state.ρ?

Reply:
The initialization state.ρ = aux_ref_state.ρ is not hardcoded for now, and so could be problematic, see issue #2111 . If it is initialized, I do not expect that there would be a change, but choosing the ref is safer since that one is not updated. We can later on ensure that the the state.ρ stays constant.

Reply:
I think there might be a issue that your density is not the reference density but you use reference density to compute the flux

Reply:
@Zhengyu-Huang can you define what you mean by _your density_ in terms of variables in the code? I do not understand what you mean. Please clarify the possible problem so that we can resolve it.